service/controller/resource/nodepool/template/main.json

```diff
@@ -316,24 +322,19 @@
                 "storageAccountType": "[variables('storageAccountType')]"
               }
             },
-            "dataDisks": [
-              {
-                "caching": "ReadWrite",
-                "createOption": "Empty",
-                "diskSizeGB": "[parameters('dockerVolumeSizeGB')]",
-                "managedDisk": {
-                  "storageAccountType": "[variables('storageAccountType')]"
-                },
-                "lun": 21
-              },
+            "copy": [
               {
-                "caching": "ReadWrite",
-                "createOption": "Empty",
-                "diskSizeGB": "[parameters('kubeletVolumeSizeGB')]",
-                "managedDisk": {
-                  "storageAccountType": "[variables('storageAccountType')]"
-                },
-                "lun": 22
+                "name": "dataDisks",
+                "count": "[length(parameters('dataDisks'))]",
+                "input": {
```

Comment:
Should there be `"name": "[copyIndex('dataDisks').NameSuffix]",` inside `input`?

Reply:
We didn't name the disks in the past, and @whites11 told me the name is probably ignored when creating disks from VMSS. We can still set it if you think it's best.